File: leetcode/28.实现 strStr()/28.实现 strStr().py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        n_len = len(needle)
        if n_len == 0:
            return 0
        for i in range(len(haystack)):
            if haystack[i] == needle[0]:
                h_str = haystack[i:i + n_len]
                if h_str == needle:
                    return i
        return -1


if __name__ == '__main__':
    s1 = "hello"
    s2 = "ll"
    s1 = ''
    s2 = ''
    print(Solution().strStr(s1, s2))

File: leetcode/28.实现 strStr()/28.实现 strStr().go
package main

func strStr(haystack string, needle string) int {
	n := len(needle)
	h := len(haystack)
	if n == 0 {
		return n
	}
	for i := 0; i < h; i++ {
		if haystack[i] == needle[0] {
			if i+n > h {
				return -1
			}
			if haystack[i:i+n] == needle {
				return i
			}
		}
	}
	return -1
}
func main() {
	s1 := "hello"
	s2 := "ll"
	println(strStr(s1, s2))
}
